pyiron_base/project/generic.py

```diff
@@ -1573,8 +1572,7 @@ def __init__(self):
         initialize the flag self._check_postgres, to control whether pyiron is
         set to communicate with a postgres database.
         """
-        s = Settings()
-        connection_string = s._configuration['sql_connection_string']
+        connection_string = state.settings._configuration['sql_connection_string']
```

Comment:
I suggest to use `state.database.sql_connection_string`, but if `state.setting` should be kept, I noticed there are both `state.setting.configuration` and `state.setting._configuration`. I think we should use the former.

Reply:
Yep, I agree with your entire chain of logic/preferences. It got fixed dealing with your earlier comment though so no further action here.